src/main/webapp/app/exam/participate/exam-participation.service.ts

```diff
@@ -1,9 +1,145 @@
 import { Injectable } from '@angular/core';
 import { SERVER_API_URL } from 'app/app.constants';
+import { Observable } from 'rxjs';
+import { StudentExam } from 'app/entities/student-exam.model';
+import { HttpClient } from '@angular/common/http';
+import { LocalStorageService } from 'ngx-webstorage';
+import { Submission } from 'app/entities/submission.model';
+import { ModelingSubmissionService } from 'app/exercises/modeling/participate/modeling-submission.service';
+import { ProgrammingSubmissionService } from 'app/exercises/programming/participate/programming-submission.service';
+import { TextSubmissionService } from 'app/exercises/text/participate/text-submission.service';
+import { FileUploadSubmissionService } from 'app/exercises/file-upload/participate/file-upload-submission.service';
+import { Exercise, ExerciseType } from 'app/entities/exercise.model';
+import { ModelingSubmission } from 'app/entities/modeling-submission.model';
+import { tap } from 'rxjs/operators';
+import { TextSubmission } from 'app/entities/text-submission.model';
 
 @Injectable({ providedIn: 'root' })
 export class ExamParticipationService {
-    public resourceUrl = SERVER_API_URL + 'api/courses';
+    private resourceUrl = SERVER_API_URL + 'api/courses';
+    private studentExam: StudentExam;
+    private submissionSaveList: Submission[] = [];
```

Comment:
I think we should not store the submissions separately. They are already part of studentExam.exercises.participations.submissions

Reply:
Those are the submissions that we synchronize back to the server. I am reusing the submissionServices from other components -> they will be synchronized synchronized separately. But you are right it is not good to keep multiple references to the same object, because some may become outdated, I will look for another way (maybe a list of submissionIds)